core/types/src/key_request.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+
+//! SGX key request rust types
+
+use crate::{new_type_wrapper, Attributes, MiscellaneousSelect};
+use mc_sgx_core_sys_types::{
+    sgx_config_svn_t, sgx_cpu_svn_t, sgx_isv_svn_t, sgx_key_id_t, sgx_key_request_t,
+};
+
+new_type_wrapper! {
+    CpuSvn, sgx_cpu_svn_t;
+}
+
+impl Default for CpuSvn {
+    fn default() -> Self {
+        CpuSvn(sgx_cpu_svn_t { svn: [0; 16] })
+    }
+}
+
+new_type_wrapper! {
+    KeyRequest, sgx_key_request_t;
+}
+
+impl Default for KeyRequest {
+    fn default() -> Self {
+        KeyRequest(sgx_key_request_t {
+            key_name: 0,
+            key_policy: 0,
+            isv_svn: 0,
+            reserved1: 0,
+            cpu_svn: CpuSvn::default().into(),
+            attribute_mask: Attributes::default().into(),
+            // TODO ideally this should be random
+            key_id: sgx_key_id_t { id: [0; 32] },
+            misc_mask: MiscellaneousSelect::default().into(),
+            config_svn: 0,
+            reserved2: [0; 434],
+        })
+    }
+}
+
+impl KeyRequest {
+    /// Set the key name to use in the key request
+    ///
+    /// # Arguments
+    ///
+    /// * `key_name`: The key name to use
+    pub fn set_key_name(mut self, key_name: u16) -> Self {
```

Comment:
wasn't sure if we wanted to go this far into the types, or if we just wanted more simple newtype wrappers and not worry about how easy it is to construct or manipulate the type at this time.